src/math.rs

```diff
@@ -67,3 +67,137 @@ impl Cube {
         )
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct OrientedBeam {
```

Comment:
Do you think it makes sense to specify the OrientedBeam as similar as possible as https://docs.rs/collision/0.20.1/collision/struct.Obb.html ? This would mean that we need a Vector2 for the extents, a Vector3 as axis, and a Point3 as some start/center/translation...

Reply:
That's a good question. A minimal representation would need even one coordinate less; e.g. the xy position of the world origin in beam coordinates, but probably harder to specify in the client. It's also not clear to me how the Obb represents rotations around the `axis`. Vector length?

If we're optimizing for computational efficiency, I should store the conjugate of the rotation, or the inverse transformation matrix. Do you think I should have different proto and Rust representations to achieve both?

Reply:
Hm, I think the cleanest at the moment would be just to change
```rust
    pub rotation: Quaternion<f64>,
    pub translation: Vector3<f64>,
    pub half_extent_x: f64,
    pub half_extent_y: f64,
```
to
```rust
    pub rotation: Quaternion<f64>,
    pub translation: Point3<f64>,
    pub extents: Vector2<f64>,
```
What do you think?

Reply:
I have no objections to the extent being a Vector2 but I'm unsure why the translation should be a Point3 instead of a Vector3.

Reply:
This would be in accordance with the cgmath and collision libs, where (as far as I understand it) a point has absolute Euclidean coordinates, whereas a vector is just a vector defining relative coordinates.

Reply:
Yes, and I think we want it to be a vector. That's the type that's used to transform Point3s, which is the type of points in the point cloud. See e.g. https://docs.rs/cgmath/0.17.0/cgmath/prelude/trait.Transform3.html

Reply:
Isn't `translation` the starting position of the oriented beam box?

Reply:
Right, both viewpoints are valid. I find it easier to imagine translation + rotation as the transform that takes local coordinates, wherein the beam is vertical, to global coordinates. This is also the viewpoint of my use case for this query. There's also an argument that since that is also how the translation will be used (i.e. for transforming points) this representation will save an additional cast.

So, I'd suggest we go with a transform Decomposed<Vector3, Quaternion> that represents the transform from global into local ("beam") coordinates, simply because that direction will not require an additional inversion when transforming each point. Perhaps we can also make the fields private if you think there is a danger of them being misinterpreted.

Reply:
[`Decomposed`](https://docs.rs/cgmath/0.17.0/cgmath/struct.Decomposed.html) also includes a scaling factor, which we don't need. Then I would rather stick with a `Vector3` and a `Quaternion`.